requirements/README.md

```diff
@@ -1,36 +1,18 @@
-The requirements.txt and requirements_ansible.txt files are generated from requirements.in and requirements_ansible.in, respectively, using `pip-tools` `pip-compile`. The following commands should do this if ran inside the tools_awx container.
+The `requirements.txt` and `requirements_ansible.txt` files are generated from `requirements.in` and `requirements_ansible.in`, respectively, using `pip-tools` `pip-compile`. The following commands should do this if ran inside the tools_awx container.
 
-Run these commands from the root of the awx repo. This will produce python 3 requirements files.
+Run `./upgrade.sh` command from inside `./requirements` directory of the awx repo.
 
-```
-python3 -m venv /buildit
-source /buildit/bin/activate
-pip install pip-tools
-pip install pip --upgrade
-
-pip-compile -U -r --allow-unsafe --output-file requirements/requirements.txt requirements/requirements.in
-pip-compile -U -r --allow-unsafe --output-file requirements/requirements_ansible_py3.txt requirements/requirements_ansible.in
-```
+This script will:
 
-Remove the `docutils` line from `requirements/requirements.txt`.
+  - Update `requirements.txt` based on `requirements_ansible.in` and `requirements.in`
+  - Update/generate `requirements_ansible.txt`
+    - including an automated patch that adds `python_version < "3"` for Python 2 backwarts-compatible scritps
+  - Removes the `docutils` dependency line from `requirements.txt` and `requirements_ansible.txt`.
 
-The Ansible venv requirements file needs to start with the python 2 version
-as a base. Then we can run the tool again to get the python 3 version.
-Consult the output of the `diff` command and add a conditional switch in those cases.
-
-```
-virtualenv -p python2 /buildit_py2
-source /buildit_py2/bin/activate
-pip install pip-tools
-pip install pip --upgrade
-
-pip-compile -U -r --allow-unsafe --output-file requirements/requirements_ansible.txt requirements/requirements_ansible.in
-diff requirements/requirements_ansible_py3.txt requirements/requirements_ansible.txt
-rm requirements/requirements_ansible_py3.txt
-```
+You can also upgrade the dependencies by running `./upgrade.sh upgrade`
 
-Python 3 exceptions should be added to relevant `requirements_ansible.txt` lines
-after version numbers with the syntax of `; python_version < '3'`.
+Known issue:
+  - Script will tangle the `This file is autogenerated by pip-compile`-part. Please exclude this on commits.
```

Comment:
This is about the comment in the file, right? I don't know if it's even important enough to put in these docs. It's perfectly clear that some runs will use different flags. I think it's justifiable to say that we're using the tool in the intended way to let it show whatever the last invocation was. Otherwise maybe we could find some flag that omits the header comments, but it's just not important to me either way.

Reply:
Exactly. I'll add the pip-compile `--no-header` flag.

Reply:
Added the `--no-header` flag https://github.com/ansible/awx/pull/4123/commits/0ccb85417e56b03b76b91b5341c2410e60b0936c